# Cars Analysis

In [1]:
# Load Packages
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings("ignore")

# Load Data Frame

In [2]:
cars = pd.read_csv("Cars Data.csv")
cars.head()

Make           Model   Type Origin DriveTrain      MSRP   Invoice  \
0  Acura             MDX    SUV   Asia        All  $36,945   $33,337    
1  Acura  RSX Type S 2dr  Sedan   Asia      Front  $23,820   $21,761    
2  Acura         TSX 4dr  Sedan   Asia      Front  $26,990   $24,647    
3  Acura          TL 4dr  Sedan   Asia      Front  $33,195   $30,299    
4  Acura      3.5 RL 4dr  Sedan   Asia      Front  $43,755   $39,014    

   EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0         3.5        6.0         265        17           23    4451   
1         2.0        4.0         200        24           31    2778   
2         2.4        4.0         200        22           29    3230   
3         3.2        6.0         270        20           28    3575   
4         3.5        6.0         225        18           24    3880   

   Wheelbase  Length  
0        106     189  
1        101     172  
2        105     183  
3        108     186  
4        115     197

# Data Understanding

In [3]:
cars.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 428 entries, 0 to 427
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Make         428 non-null    object 
 1   Model        428 non-null    object 
 2   Type         428 non-null    object 
 3   Origin       428 non-null    object 
 4   DriveTrain   428 non-null    object 
 5   MSRP         428 non-null    object 
 6   Invoice      428 non-null    object 
 7   EngineSize   428 non-null    float64
 8   Cylinders    426 non-null    float64
 9   Horsepower   428 non-null    int64  
 10  MPG_City     428 non-null    int64  
 11  MPG_Highway  428 non-null    int64  
 12  Weight       428 non-null    int64  
 13  Wheelbase    428 non-null    int64  
 14  Length       428 non-null    int64  
dtypes: float64(2), int64(6), object(7)
memory usage: 50.3+ KB


In [4]:
cars.describe()

EngineSize   Cylinders  Horsepower    MPG_City  MPG_Highway  \
count  428.000000  426.000000  428.000000  428.000000   428.000000   
mean     3.196729    5.807512  215.885514   20.060748    26.843458   
std      1.108595    1.558443   71.836032    5.238218     5.741201   
min      1.300000    3.000000   73.000000   10.000000    12.000000   
25%      2.375000    4.000000  165.000000   17.000000    24.000000   
50%      3.000000    6.000000  210.000000   19.000000    26.000000   
75%      3.900000    6.000000  255.000000   21.250000    29.000000   
max      8.300000   12.000000  500.000000   60.000000    66.000000   

            Weight   Wheelbase      Length  
count   428.000000  428.000000  428.000000  
mean   3577.953271  108.154206  186.362150  
std     758.983215    8.311813   14.357991  
min    1850.000000   89.000000  143.000000  
25%    3104.000000  103.000000  178.000000  
50%    3474.500000  107.000000  187.000000  
75%    3977.750000  112.000000  194.000000  
max    7190.000000  144.000000  238.000000

In [5]:
cars.isnull().sum()

Make           0
Model          0
Type           0
Origin         0
DriveTrain     0
MSRP           0
Invoice        0
EngineSize     0
Cylinders      2
Horsepower     0
MPG_City       0
MPG_Highway    0
Weight         0
Wheelbase      0
Length         0
dtype: int64

In [6]:
cars.columns

Index(['Make', 'Model', 'Type', 'Origin', 'DriveTrain', 'MSRP', 'Invoice',
       'EngineSize', 'Cylinders', 'Horsepower', 'MPG_City', 'MPG_Highway',
       'Weight', 'Wheelbase', 'Length'],
      dtype='object')

In [7]:
cars.shape

(428, 15)

# Data PreProcessing

# 1. Find all Null Value in the dataset. If there is any null value in any column, then fill it with the mean of that column.

In [8]:
# Feeling the null vlaues
cars["Cylinders"].fillna(cars["Cylinders"].mean(), inplace=True)

In [9]:
cars.isnull().sum()

Make           0
Model          0
Type           0
Origin         0
DriveTrain     0
MSRP           0
Invoice        0
EngineSize     0
Cylinders      0
Horsepower     0
MPG_City       0
MPG_Highway    0
Weight         0
Wheelbase      0
Length         0
dtype: int64

# 2. Check what are the different types of "Make" are there in our dataset. And, what is the count (occurrence) of each "Make" in the data?

In [10]:
cars["Make"].value_counts()

Make
Toyota           28
Chevrolet        27
Mercedes-Benz    26
Ford             23
BMW              20
Audi             19
Nissan           17
Honda            17
Chrysler         15
Volkswagen       15
Mitsubishi       13
Dodge            13
Volvo            12
Hyundai          12
Jaguar           12
Subaru           11
Kia              11
Pontiac          11
Mazda            11
Lexus            11
Buick             9
Mercury           9
Lincoln           9
Cadillac          8
GMC               8
Saturn            8
Suzuki            8
Infiniti          8
Acura             7
Saab              7
Porsche           7
Land Rover        3
Oldsmobile        3
Jeep              3
Isuzu             2
MINI              2
Scion             2
Hummer            1
Name: count, dtype: int64

# 3. Show all the records where "Origin" is Asia or Europe.

In [11]:
cars["Origin"].value_counts()

Origin
Asia      158
USA       147
Europe    123
Name: count, dtype: int64

In [12]:
cars[(cars["Origin"] == "Asia") | (cars["Origin"] == "Europe")]

Make                    Model   Type  Origin DriveTrain      MSRP  \
0    Acura                      MDX    SUV    Asia        All  $36,945    
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
..     ...                      ...    ...     ...        ...       ...   
423  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
424  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
425  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
426  Volvo                      V40  Wagon  Europe      Front  $26,135    
427  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
0    $33,337          3.5        6.0         265        17           23   
1    $21,761          2.0        4.0         200        24           31   
2    $24,647          2.4        4.0         200        22           29   
3    $30,299          3.2        6.0         270        20           28   
4    $39,014          3.5        6.0         225        18           24   
..        ...         ...        ...         ...       ...          ...   
423  $38,203          2.4        5.0         197        21           28   
424  $40,083          2.3        5.0         242        20           26   
425  $42,573          2.9        6.0         268        19           26   
426  $24,641          1.9        4.0         170        22           29   
427  $33,112          2.5        5.0         208        20           27   

     Weight  Wheelbase  Length  
0      4451        106     189  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
..      ...        ...     ...  
423    3450        105     186  
424    3450        105     186  
425    3653        110     190  
426    2822        101     180  
427    3823        109     186  

[281 rows x 15 columns]

# 4. Remove all the records (rows) where "Weight" is above 4000.

In [13]:
cars[(cars["Weight"] <= 4000)]

Make                    Model   Type  Origin DriveTrain      MSRP  \
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
5    Acura  3.5 RL w/Navigation 4dr  Sedan    Asia      Front  $46,100    
..     ...                      ...    ...     ...        ...       ...   
423  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
424  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
425  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
426  Volvo                      V40  Wagon  Europe      Front  $26,135    
427  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
1    $21,761          2.0        4.0         200        24           31   
2    $24,647          2.4        4.0         200        22           29   
3    $30,299          3.2        6.0         270        20           28   
4    $39,014          3.5        6.0         225        18           24   
5    $41,100          3.5        6.0         225        18           24   
..        ...         ...        ...         ...       ...          ...   
423  $38,203          2.4        5.0         197        21           28   
424  $40,083          2.3        5.0         242        20           26   
425  $42,573          2.9        6.0         268        19           26   
426  $24,641          1.9        4.0         170        22           29   
427  $33,112          2.5        5.0         208        20           27   

     Weight  Wheelbase  Length  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
5      3893        115     197  
..      ...        ...     ...  
423    3450        105     186  
424    3450        105     186  
425    3653        110     190  
426    2822        101     180  
427    3823        109     186  

[325 rows x 15 columns]

# 5. Increase all the values of "MPG_City" column by 3.

In [14]:
cars["MPG_City"] = cars["MPG_City"] + 3
cars

Make                    Model   Type  Origin DriveTrain      MSRP  \
0    Acura                      MDX    SUV    Asia        All  $36,945    
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
..     ...                      ...    ...     ...        ...       ...   
423  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
424  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
425  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
426  Volvo                      V40  Wagon  Europe      Front  $26,135    
427  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
0    $33,337          3.5        6.0         265        20           23   
1    $21,761          2.0        4.0         200        27           31   
2    $24,647          2.4        4.0         200        25           29   
3    $30,299          3.2        6.0         270        23           28   
4    $39,014          3.5        6.0         225        21           24   
..        ...         ...        ...         ...       ...          ...   
423  $38,203          2.4        5.0         197        24           28   
424  $40,083          2.3        5.0         242        23           26   
425  $42,573          2.9        6.0         268        22           26   
426  $24,641          1.9        4.0         170        25           29   
427  $33,112          2.5        5.0         208        23           27   

     Weight  Wheelbase  Length  
0      4451        106     189  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
..      ...        ...     ...  
423    3450        105     186  
424    3450        105     186  
425    3653        110     190  
426    2822        101     180  
427    3823        109     186  

[428 rows x 15 columns]

# Conclusion

# Conclusion – Cars Dataset Analysis

# 1. Data Cleaning & Preparation

    i. The dataset originally had 428 car records and 15 features.

    ii. Missing values in the Cylinders column were handled by filling them with the column mean.

    iii. Text-based columns like Make, Model, and Origin were categorical, while numerical columns such as Horsepower, Weight, and MPG were continuous.

# 2. Exploratory Insights

    i. The dataset contains cars from multiple manufacturers (Make) with varying frequencies, highlighting the diversity of the automobile market.

    ii. Filtering by Origin showed that Asian and European cars make up a significant portion of the dataset, indicating strong global representation.

    iii. Removing heavy cars (Weight > 4000 lbs) ensured that the analysis focused on more common, consumer-level vehicles rather than specialized or oversized ones.

# 3. Fuel Efficiency Adjustment

    i. After increasing MPG_City values by 3, the dataset reflected an improvement in city fuel efficiency, which could represent hypothetical advancements in car technology or testing an alternative scenario.

# 4. Key Takeaways

    i. The dataset is well-structured for car performance analysis (engine size, horsepower, MPG, weight, etc.).

    ii. Cleaning and preprocessing steps significantly improved data quality.

    iii. Manufacturer frequency and regional origin insights are useful for understanding market distribution.

    iv. Adjusting MPG_City highlights how policy or engineering improvements could impact efficiency studies.

# Overall, the analysis prepares the dataset for further modeling or visualization, such as predicting car prices (MSRP) or clustering vehicles by specifications.